transport/controller.cc

```diff
@@ -112,47 +114,61 @@ future<> controller::do_start_server() {
         std::vector<listen_cfg> configs;
         int native_port_idx = -1, native_shard_aware_port_idx = -1;
 
-        if (cfg.native_transport_port.is_set() ||
-                (!cfg.native_transport_port_ssl.is_set() && !cfg.native_transport_port.is_set())) {
-            // Non-SSL port is specified || neither SSL nor non-SSL ports are specified
-            configs.emplace_back(listen_cfg{ socket_address{ip, cfg.native_transport_port()}, false });
-            _listen_addresses.push_back(configs.back().addr);
-            native_port_idx = 0;
-        }
-        if (cfg.native_shard_aware_transport_port.is_set() ||
-                (!cfg.native_shard_aware_transport_port_ssl.is_set() && !cfg.native_shard_aware_transport_port.is_set())) {
-            configs.emplace_back(listen_cfg{ socket_address{ip, cfg.native_shard_aware_transport_port()}, true });
-            _listen_addresses.push_back(configs.back().addr);
-            native_shard_aware_port_idx = native_port_idx + 1;
-        }
-
-        // main should have made sure values are clean and neatish
-        if (utils::is_true(utils::get_or_default(ceo, "enabled", "false"))) {
-            auto cred = std::make_shared<seastar::tls::credentials_builder>();
-            utils::configure_tls_creds_builder(*cred, std::move(ceo)).get();
-
-            logger.info("Enabling encrypted CQL connections between client and server");
+        if (!_maintenance_port) {
```

Comment:
I'm confused at all these changes. I expected this maintenance port to be a pure addition, but it looks like it is exclusive with the regular CQL port? I don't think that makes sense. The regular CQL port can already be disabled via config and nodetool, the maintenance socket should be just another optional way for CQL traffic to be received.

BTW, maintenance_port is called maintenance_socket elsewhere, please use consistent naming.

Reply:
If I understand correctly, there will be two instances of `cql_server_controller`, one corresponding to regular port, one to maintenance port.

The maintenance port one has a bunch of stuff done differently / disabled (like authentication).

> but it looks like it is exclusive with the regular CQL port? 

Why? Doesn't seem like it to me

Reply:
Ah, it is really hard to see what is going an due to all the changes in this method.

Reply:
There are not many actual changes, most of the diff comes from changing the indentation, IIUC. @margdoc please split the indentation change to a separate commit